core/client/fs/src/main/java/alluxio/client/file/policy/LocalFirstPolicy.java

```diff
@@ -63,23 +84,33 @@ public WorkerNetAddress getWorkerForNextBlock(Iterable<BlockWorkerInfo> workerIn
       long blockSizeBytes) {
     List<BlockWorkerInfo> shuffledWorkers = Lists.newArrayList(workerInfoList);
     Collections.shuffle(shuffledWorkers);
-    List<WorkerNetAddress> enoughCapacityWorkers = shuffledWorkers.stream()
+    // Workers must have enough capacity to hold the block.
+    List<BlockWorkerInfo> candidateWorkers = shuffledWorkers.stream()
         .filter(worker -> worker.getCapacityBytes() >= blockSizeBytes)
-        .map(BlockWorkerInfo::getNetAddress)
         .collect(Collectors.toList());
+    if (mAvoidEviction) {
+      // Workers must have enough availability to hold the block.
+      List<BlockWorkerInfo> nonEvictingWorkers = candidateWorkers.stream()
+          .filter(worker -> getAvailableBytes(worker) >= blockSizeBytes)
+          .collect(Collectors.toList());
+      if (!nonEvictingWorkers.isEmpty()) {
+        candidateWorkers = nonEvictingWorkers;
+      }
```

Comment:
Would we be able to reuse local first policy and in the avoid eviction class filter the `workerInfoList` before passing it to the local first policy's `getWorkerForNextBlock`?

Reply:
good call, much cleaner now